cpp/perspective/src/cpp/emscripten.cpp

```diff
@@ -1413,34 +1413,53 @@ namespace binding {
              * columns on the View.
              */
             std::vector<t_dtype> input_types(input_columns.size());
+
+            // If the input columns are invalid, an error will be thrown here.
             for (auto i = 0; i < input_columns.size(); ++i) {
                 input_types[i] = schema->get_dtype(input_columns[i]);
             }
 
             t_computation computation = t_computed_column::get_computation(
                 computed_function_name, input_types);
-            
+        
+            // Throw an exception if the computation is invalid - the UI will
+            // prevent users from saving invalidly-typed computations.
             if (computation.m_name == INVALID_COMPUTED_FUNCTION) {
```

Comment:
`INVALID_COMPUTED_FUNCTION` magic values seems to indicate to me this error should occur in `get_computation()`, since we're already using `ABORT()` for error messages?

Reply:
The function returns `INVALID_COMPUTED_FUNCTION` instead of abort so that the calling function can print a verbose error message, but as discussed offline we should get rid of the verbose error messages along with other refactors in a debt relief task.